### PR TITLE
#RI-5138 - add dry run job panel

### PR DESCRIPTION
**Pull request by AmirAllayarovSofteq**

### CODE REVIEW

File: redisinsight/ui/src/slices/rdi/dryRun.tsx [hunk 1]
```diff
@@ -0,0 +1,81 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit'
+import { AxiosError } from 'axios'
+import { apiService, } from 'uiSrc/services'
+import { addErrorNotification } from 'uiSrc/slices/app/notifications'
+import { getApiErrorMessage, isStatusSuccessful } from 'uiSrc/utils'
+
+import { AppDispatch, RootState } from '../store'
+
+export const initialState: any = {
```
Comment (romansergeenkosofteq) [inline redisinsight/ui/src/slices/rdi/dryRun.tsx:9]: please add interface

Comment (AmirAllayarovSofteq) [reply redisinsight/ui/src/slices/rdi/dryRun.tsx]: thanks

File: redisinsight/ui/src/pages/rdi/pipeline/components/dry-run-job-commands/DryRunJobCommands.tsx [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+import { rdiDryRunJobSelector } from 'uiSrc/slices/rdi/dryRun'
+import { MonacoLanguage } from 'uiSrc/constants'
+
+const DryRunJobCommands = () => {
+  const { results } = useSelector(rdiDryRunJobSelector)
+
+  const [commands, setCommands] = useState('')
+
+  useEffect(() => {
+    if (results && results.commands?.status === 'success') {
```
Comment (kyle-marcum) [inline redisinsight/ui/src/pages/rdi/pipeline/components/dry-run-job-commands/DryRunJobCommands.tsx:16]: Screenshot 2024-01-09 at 2 23 03 PM

Should `commands` and `status` be added to the `IPipeline` interface?

File: redisinsight/ui/src/pages/rdi/pipeline/pages/jobs/Jobs.spec.tsx [hunk 3]
```diff
@@ -82,4 +83,29 @@ describe('Jobs', () => {
 
     expect(screen.getByTestId('rdi-pipeline-transformation-link')).toHaveAttribute('href', 'https://docs.redis.com/latest/rdi/data-transformation/data-transformation-pipeline/')
   })
+
+  it('should send telemetry event with proper data', () => {
```
Comment (kyle-marcum) [inline redisinsight/ui/src/pages/rdi/pipeline/pages/jobs/Jobs.spec.tsx:87]: There is also a `TelemetryPageView.RDI_JOBS` event that does seem to have a unit test. Doesn't look like the event was added with this PR but writing a test for it should be quick and easy.

**Comment by kyle-marcum** [comment]
I left a few comments regarding typescript types. What sort of IDE do you use? I know that vscode has a plugin for typescript that will surface the type problems in your editor so that they are easy to see.

File: redisinsight/ui/src/pages/rdi/pipeline/components/dry-run-job-commands/DryRunJobCommands.tsx (hunk 2 above)
Comment (AmirAllayarovSofteq) [reply redisinsight/ui/src/pages/rdi/pipeline/components/dry-run-job-commands/DryRunJobCommands.tsx]: Thank you. I updated interface

File: redisinsight/ui/src/pages/rdi/pipeline/components/dry-run-job-transformations/DryRunJobTransformations.tsx [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+import { rdiDryRunJobSelector } from 'uiSrc/slices/rdi/dryRun'
+import MonacoJson from 'uiSrc/components/monaco-editor/components/monaco-json'
+
+const DryRunJobTransformations = () => {
+  const { results } = useSelector(rdiDryRunJobSelector)
+
+  const [transformations, setTransformations] = useState('')
+
+  useEffect(() => {
+    if (results && results.transformations?.status === 'success') {
```
[the remark this answers is not in the document]
Comment (AmirAllayarovSofteq) [reply redisinsight/ui/src/pages/rdi/pipeline/components/dry-run-job-transformations/DryRunJobTransformations.tsx]: Thank you

File: redisinsight/ui/src/pages/rdi/pipeline/pages/jobs/Jobs.spec.tsx (hunk 3 above)
Comment (AmirAllayarovSofteq) [reply redisinsight/ui/src/pages/rdi/pipeline/pages/jobs/Jobs.spec.tsx:87]: Probably you miss it, please, take a look at 29 line
`  it('should call proper sendPageViewTelemetry', () => {`

Comment (kyle-marcum) [reply redisinsight/ui/src/pages/rdi/pipeline/pages/jobs/Jobs.spec.tsx:87]: I did miss it. My mistake.

**Comment by AmirAllayarovSofteq** [comment]
> I left a few comments regarding typescript types. What sort of IDE do you use? I know that vscode has a plugin for typescript that will surface the type problems in your editor so that they are easy to see.

I use WebStorm (but previously i  used VS code, i missed ts problems because i'm updated interfaces in last commits and forgot to take a look at other files...) My bad, thank you :)
I hope after review rdi docs and confirm everything we can also finalized our types and interfaces

**Review by romansergeenkosofteq — APPROVED** [review]

**Pull request merged**